ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/text_embedding/TextEmbeddingModel.java

```diff
@@ -0,0 +1,239 @@
+                engine
+        );
+    }
+
+    @Override
+    public void close() {
+        if (modelHelper != null && modelId != null) {
+            modelHelper.deleteFileCache(modelId);
+            if (predictor != null) {
+                log.debug("close predictor for model {}", modelId);
+                predictor.close();
+                predictor = null;
+            }
+            if (model != null) {
+                log.debug("close model for model {}", modelId);
+                model.close();
+                model = null;
+            }
+        }
+    }
+
+    protected void loadTextEmbeddingModel(File modelZipFile, String modelId, String modelName, FunctionName functionName, Integer version,
+                                       MLModelConfig modelConfig,
+                                       String engine) {
+        try {
+            if (FunctionName.TEXT_EMBEDDING != functionName) {
+                throw new IllegalArgumentException("wrong function name");
+            }
+            if (!PYTORCH_ENGINE.equals(engine) && !ONNX_ENGINE.equals(engine)) {
+                throw new IllegalArgumentException("unsupported engine");
+            }
+            AccessController.doPrivileged((PrivilegedExceptionAction<Void>) () -> {
+                ClassLoader contextClassLoader = Thread.currentThread().getContextClassLoader();
+                try {
+                    System.setProperty("PYTORCH_PRECXX11", "true");
+                    System.setProperty("DJL_CACHE_DIR", DJL_CACHE_PATH.toAbsolutePath().toString());
+                    System.setProperty("java.library.path", DJL_CACHE_PATH.toAbsolutePath().toString());
+                    Thread.currentThread().setContextClassLoader(ai.djl.Model.class.getClassLoader());
+                    Path modelPath = getModelCachePath(modelId, modelName, version);
+                    File pathFile = new File(modelPath.toUri());
+                    if (pathFile.exists()) {
+                        FileUtils.deleteDirectory(pathFile);
+                    }
+                    ZipUtils.unzip(new FileInputStream(modelZipFile), modelPath);
+                    boolean findModelFile = false;
+                    for (File file : pathFile.listFiles()) {
+                        String name = file.getName();
+                        if (name.endsWith(PT) || name.endsWith(ONNX)) {
+                            if (findModelFile) {
+                                throw new IllegalArgumentException("found multiple models");
+                            }
+                            findModelFile = true;
+                            String suffix = name.substring(name.lastIndexOf("."));
+                            file.renameTo(new File(modelPath.resolve(modelName + suffix).toUri()));
+                        }
+                    }
+                    Map<String, Object> arguments = new HashMap<>();
+                    arguments.put("engine", engine);
```

Comment:
You don't need use arguments, you can just use:

```
Criteria.builder().optEngine(engine)
```

Reply:
Good point, will fix